corehq/util/models.py

```diff
@@ -289,8 +287,20 @@ def __set_name__(self, owner, name):
     def __get__(self, obj, objtype=None):
         if obj is None:
             return self
-        fobj = getattr(obj, self.fk.name)
-        return fobj.value if fobj is not None else None
+        fobj_id = getattr(obj, f"{self.fk.name}_id")
+        return self.get_value(fobj_id)
+
+    @cached_property
+    def get_value(self):
+        def get_value(fk_id):
+            try:
+                return manager.filter(pk=fk_id).values_list('value', flat=True)[0]
```

Comment:
I assume [it's fine](https://code.djangoproject.com/ticket/9619), but raising the question here for completeness: I forget if this query technique will call `Field.to_python` before returning the value? For example, will this work with `NullJsonField`? (Not that that is a good idea since the semantics around mutating the value will likely be confusing.)